DatadogSessionReplay/Sources/DatadogCoreIntegration/SRContextPublisher.swift

```diff
@@ -15,11 +15,19 @@ internal class SRContextPublisher {
         self.core = core
     }
 
-    /// Notifies other Features on the state of  Session Replay recording.
-    func setRecordingIsPending(_ value: Bool) {
-        core?.set(
+    /// Notifies other Features if Session Replay is recording.
+    func setHasReplay(_ value: Bool) {
+        core?.update(
             feature: RUMDependency.srBaggageKey,
             attributes: { [RUMDependency.hasReplay: value] }
         )
     }
+
+    /// Notifies other Features on the state of Session Replay records count.
+    func setRecordsCount(_ value: [String: Int64]) {
```

Comment:
suggestion/ The `value: [String: Int64]` sounds very ambiguous in this context. We need to track the code to the very bottom to understand that `String` is `view.id`. We could name it better `_ recordsCountByViewID:` or annotate properly with the method comment.